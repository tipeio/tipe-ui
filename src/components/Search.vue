<template>
  <input @click="onClick">
</template>

<script>
export default {
  name: 'TipeSearch',

  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>

<style scoped>
button {
  border: '1px solid #eee';
  border-radius: 3;
  background-color: '#FFFFFF';
  cursor: 'pointer';
  font-size: 15;
  padding: '3px 10px';
  margin: 10;
}
</style>
