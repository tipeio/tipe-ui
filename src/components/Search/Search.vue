<template>
  <div :data-tipe-ui="$options.name">
    <type-icon
      class="search-icon"
      icon="search"
      color="#b5baca"
    />
    <input
      type="search"
      placeholder="Find everthing..."
    >
  </div>
</template>

<script>
import TypeIcon from '../Icon'

export default {
  name: 'TipeSearch',
  components: {
    TypeIcon
  },
  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeSearch'] {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: auto;
}

.search-icon {
  flex: 0 0 auto;
  height: 1.25rem;
  width: 1.25rem;
}

[data-tipe-ui='TipeSearch'] > input {
  flex: 1;
  margin-left: 0.437rem;
  margin-right: 0.437rem;
  border: none;
  border-radius: 0;
  color: var(--text-gray);
  background-color: transparent;
  font-size: 1rem;
  height: 1.625rem;
  line-height: 1.625rem;
}

[data-tipe-ui='TipeSearch'] > input::-webkit-input-placeholder {
  color: var(--text-gray);
  opacity: 0.4;
}
</style>
