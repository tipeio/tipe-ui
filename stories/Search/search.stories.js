import { storiesOf } from '@storybook/vue'
import TipeSearch from '../../src/components/Search'

storiesOf('Search', module).add('default', () => ({
  components: { TipeSearch },
  template: '<tipe-search />'
}))
