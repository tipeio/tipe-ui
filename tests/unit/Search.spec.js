import { shallowMount } from '@vue/test-utils'
import Search from '@/components/Search.vue'

describe('Search.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Search)
    expect(wrapper).toMatchSnapshot()
  })
})
