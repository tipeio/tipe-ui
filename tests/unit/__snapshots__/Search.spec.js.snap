// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue renders 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "element": <input />,
  "isFunctionalComponent": undefined,
  "isVueComponent": true,
  "options": Object {
    "attachedToDocument": false,
    "sync": true,
  },
  "version": 2.5,
  "vm": VueComponent {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <input />,
    "$emit": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_Ctor": Object {
        "1": [Function],
      },
      "_base": [Function],
      "_parentElm": null,
      "_refElm": null,
      "beforeCreate": Array [
        [Function],
      ],
      "components": Object {
        "transition": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "name": "transition",
          "render": [Function],
        },
        "transition-group": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "name": "transition-group",
          "render": [Function],
        },
      },
      "directives": Object {},
      "filters": Object {},
      "logModifiedComponents": true,
      "methods": Object {
        "onClick": [Function],
      },
      "name": "TipeSearch",
      "provide": [Function],
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "__emitted": Object {},
    "__emittedByOrder": Array [],
    "_c": [Function],
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_provided": undefined,
    "_renderProxy": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <input />,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition",
            "render": [Function],
          },
          "transition-group": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition-group",
            "render": [Function],
          },
        },
        "directives": Object {},
        "filters": Object {},
        "logModifiedComponents": true,
        "methods": Object {
          "onClick": [Function],
        },
        "name": "TipeSearch",
        "provide": [Function],
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_provided": undefined,
      "_renderProxy": [Circular],
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": undefined,
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": [Circular],
        "data": Object {
          "on": Object {
            "click": [Function],
          },
        },
        "elm": <input />,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": true,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "input",
        "text": undefined,
      },
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
      "onClick": [Function],
    },
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": undefined,
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": [Circular],
      "data": Object {
        "on": Object {
          "click": [Function],
        },
      },
      "elm": <input />,
      "fnContext": undefined,
      "fnOptions": undefined,
      "fnScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": false,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": "input",
      "text": undefined,
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {},
      "deps": Array [],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 1,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": true,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
    "onClick": [Function],
  },
  "vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": undefined,
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <input />,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition",
            "render": [Function],
          },
          "transition-group": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition-group",
            "render": [Function],
          },
        },
        "directives": Object {},
        "filters": Object {},
        "logModifiedComponents": true,
        "methods": Object {
          "onClick": [Function],
        },
        "name": "TipeSearch",
        "provide": [Function],
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_provided": undefined,
      "_renderProxy": VueComponent {
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$el": <input />,
        "$emit": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "_base": [Function],
          "_parentElm": null,
          "_refElm": null,
          "beforeCreate": Array [
            [Function],
          ],
          "components": Object {
            "transition": Object {
              "_Ctor": Object {
                "1": [Function],
              },
              "name": "transition",
              "render": [Function],
            },
            "transition-group": Object {
              "_Ctor": Object {
                "1": [Function],
              },
              "name": "transition-group",
              "render": [Function],
            },
          },
          "directives": Object {},
          "filters": Object {},
          "logModifiedComponents": true,
          "methods": Object {
            "onClick": [Function],
          },
          "name": "TipeSearch",
          "provide": [Function],
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "__emitted": Object {},
        "__emittedByOrder": Array [],
        "_c": [Function],
        "_data": Object {},
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": true,
        "_isVue": true,
        "_provided": undefined,
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_staticTrees": null,
        "_uid": 0,
        "_vnode": [Circular],
        "_watcher": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        "_watchers": Array [
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 1,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": true,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        ],
        "onClick": [Function],
      },
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": [Circular],
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 1,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 1,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
      "onClick": [Function],
    },
    "data": Object {
      "on": Object {
        "click": [Function],
      },
    },
    "elm": <input />,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "input",
    "text": undefined,
  },
}
`;
